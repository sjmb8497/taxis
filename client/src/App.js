import React, { useState, useEffect } from 'react';
import { getTaxis } from "./api";

function App() {
  const [driverData, setDriverData] = useState([]);
  useEffect(() => {
    async function fetch() {
      try {
        const data = await getTaxis();
        console.log('sophia')
        console.log('data', data)
        setDriverData(data);
        // console.log(driverData)
      } catch (err) {
        console.error(err);
      }
    }
    fetch()
  }, [])

  return (
    <div className="App">
      <p>hello world</p>
    </div>
  );
}

export default App;
