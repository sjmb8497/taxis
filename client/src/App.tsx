import React, { useState, useEffect } from 'react';
import { Map } from './components/Map';
import { distance } from './helpers/distance';
import { locationInterface, driverDataInterface } from './helpers/interfaces'
import { getTaxis } from "./api";
import Slider from "@material-ui/core/Slider";

const App: React.FunctionComponent = () => {
  const [driverData, setDriverData] = useState<driverDataInterface>({
    pickup_eta: 1,
    drivers: [],
  });

  const [userLocation, setUserLocation] = useState<locationInterface>({
    lat: 51.5049375,
    lng: -0.0964509
  });

  const [center, setCenter] = useState<locationInterface>({
    lat: 51.5049375,
    lng: -0.0964509
  });

  const [numberOfDrivers, setNumberOfDrivers] = useState<number>(5);
  const [currentLocation, setCurrentLocation] = useState<string>('');

  const findCurrentLocation = (latitude: number, longitude: number) => {
    setUserLocation({lat: latitude,lng: longitude});

    const singaporeDistance = distance(userLocation.lat, userLocation.lng, 1.285194, 103.8522982);
    const londonDistance = distance(userLocation.lat, userLocation.lng, 51.5049375, -0.0964509);

    if (londonDistance < singaporeDistance) {
      setCurrentLocation('London');
    } else {
      setCurrentLocation('Singapore');
    }
  }

  const findClosestOffice = () => {
      navigator.geolocation.getCurrentPosition((position) => {
        return findCurrentLocation(position.coords.latitude, position.coords.longitude);
      });
    }

  const changeCenter = () => {
    if (currentLocation === 'London') {
      setCurrentLocation('Singapore')
    } else {
      setCurrentLocation('London')
    }
  }

  const handleSlider = (event: any, newValue: any) => {
    setNumberOfDrivers(newValue);
  }

  useEffect(() => {
    async function fetch() {
      try {
        findClosestOffice();
        const data = await getTaxis(center.lat, center.lng, numberOfDrivers);
        setDriverData(data);
      } catch (err) {
        console.error(err);
      }
    }
    fetch();
    const interval = setInterval(async () => {
      fetch()
    }, 5000);
    return () => clearInterval(interval);
  }, [center, numberOfDrivers])

  useEffect(() => {
    if (currentLocation === 'London') {
      setCenter({lat: 51.5049375, lng:-0.0964509})
    } else {
      setCenter({lat: 1.285194, lng:103.8522982})
    }
  }, [currentLocation])

  const marks = [
    {value: 0,label: '0'}, {value: 5,label: '5'}, {value: 10,label: '10'}, {value: 15,label: '15'}, {value: 20,label: '20'}
  ];

  return (
    <div className="App">
      <button onClick={() => changeCenter()}>Change location</button>
      <p>Change number of taxis:</p>
      <Slider
        defaultValue={numberOfDrivers}
        step={1}
        min={0}
        max={20}
        onChange={handleSlider}
        marks={marks}
      ></Slider>
      <Map taxis={driverData} center={center}/>
    </div>
  );
}

export default App;