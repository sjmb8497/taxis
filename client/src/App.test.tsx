import { render, screen } from '@testing-library/react';
import App from './App';

const mockGeolocation = {
    getCurrentPosition: jest.fn(),
    watchPosition: jest.fn()
};
  
global.navigator.geolocation = mockGeolocation;

test('renders App', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();

});

test('renders elements within app', () => {
  render(<App />);
  expect(screen.getByText('Change number of taxis:')).toBeInTheDocument();
  expect(screen.getByText('Change location')).toBeInTheDocument();
})
