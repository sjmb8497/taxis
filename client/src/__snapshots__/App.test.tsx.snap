// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <button>
      Change location
    </button>
    <p>
      Change number of taxis:
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-marked"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 25%;"
      />
      <input
        type="hidden"
        value="5"
      />
      <span
        class="MuiSlider-mark MuiSlider-markActive"
        data-index="0"
        style="left: 0%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive"
        data-index="0"
        style="left: 0%;"
      >
        0
      </span>
      <span
        class="MuiSlider-mark MuiSlider-markActive"
        data-index="1"
        style="left: 25%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive"
        data-index="1"
        style="left: 25%;"
      >
        5
      </span>
      <span
        class="MuiSlider-mark"
        data-index="2"
        style="left: 50%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel"
        data-index="2"
        style="left: 50%;"
      >
        10
      </span>
      <span
        class="MuiSlider-mark"
        data-index="3"
        style="left: 75%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel"
        data-index="3"
        style="left: 75%;"
      >
        15
      </span>
      <span
        class="MuiSlider-mark"
        data-index="4"
        style="left: 100%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel"
        data-index="4"
        style="left: 100%;"
      >
        20
      </span>
      <span
        aria-orientation="horizontal"
        aria-valuemax="20"
        aria-valuemin="0"
        aria-valuenow="5"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
    </span>
    <div
      style="height: 100vh; width: 100vw;"
    >
      <div
        style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        />
        <div
          style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
